### PR TITLE
More secure custom authentication solution

### LINKED ISSUE #117: Auth Tokens are stored insecurely
They're in localstorage *and* session-only cookies without the proper headers set.
Apparently, this is the default for nuxt/auth.

They *should* be in `Secure`, `HttpOnly`, `SameSite=Strict` cookies only, to soften the blow of a potential XSS.

### PR DESCRIPTION
**Pull request by Geometrically**
Fixes #117

### CODE REVIEW

**Review by Redblueflame — APPROVED** [review]
LGTM

**Review by Redblueflame — COMMENTED** [review]
Just a note, if vue-persist is activated on the store, the auth store will also be stored in the localstorage.

**Comment by Geometrically** [comment]
We don't use vue-persist yet, but it may be added in the future to assist with caching. I believe you can exclude stuff from vue persist though

**Comment by Redblueflame** [comment]
Then everything is fine :)

File: middleware/auth.js [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+        )
+      )
+    }
+  })
+}
+
+function getMatchedComponents(route, matches) {
+  return [].concat(
+    ...[],
+    ...route.matched.map(function (m, index) {
```
Comment (falseresync) [inline middleware/auth.js:54]: Why not arrow functions? ie `(param) => result`

File: middleware/auth.js [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+    }
+
+    if (context.app.$cookies.get('auth-token')) {
+      const cookie = context.app.$cookies.get('auth-token')
+
+      await context.store.dispatch('auth/fetchUser', { token: cookie })
+    }
+  }
+
+  if (routeOption(context.route, 'auth', false)) return
```
Comment (falseresync) [inline middleware/auth.js:20]: What are these supposed to check? They aren’t self documenting enough, some commentary is required

File: plugins/auth.js [hunk 3]
```diff
@@ -0,0 +1,3 @@
+export default ({ store }, inject) => {
```
Comment (falseresync) [inline plugins/auth.js:1]: What for? Stores are always available. And there are Vuex store “getters” for easier access

**Review by falseresync — CHANGES_REQUESTED** [review]

File: plugins/auth.js (hunk 3 above)
Comment (Geometrically) [reply plugins/auth.js:1]: The plugin is just for a utility to make it shorter to access the auth stuff. What do you mean by Vuex store getters?

File: middleware/auth.js (hunk 2 above)
Comment (Geometrically) [reply middleware/auth.js:20]: This is taken from: https://github.com/nuxt-community/auth-module/blob/dev/src/core/middleware.ts#L6

File: middleware/auth.js (hunk 1 above)
Comment (Geometrically) [reply middleware/auth.js:54]: Could you suggest what it should look like?

**Review by Scotsguy — APPROVED** [review]
Secure, SameSite=Strict, HttpOnly cookies, LGTM. Can't comment on the js quality

**Review by falseresync — APPROVED** [review]
Lgtm

**Pull request merged**